src/nemos/basis.py

```diff
@@ -1221,10 +1577,27 @@ def __init__(self, basis1: Basis, basis2: Basis) -> None:
         self._n_input_dimensionality = (
             basis1._n_input_dimensionality + basis2._n_input_dimensionality
         )
+        self._n_basis_input = None
+        self._n_output_features = None
+        self._label = "(" + basis1.label + " + " + basis2.label + ")"
```

Comment:
We definitely need the parenthesis for the multiplication, otherwise this 

```python
a,b,c = BSplineBasis(5),BSplineBasis(5),BSplineBasis(5)
example_2 = (a + b) * c
example_3 = a + (b * c)
```

Will result in the same label. I can drop the addition one